src/H5FDs3comms.c

```diff
@@ -1645,7 +1654,7 @@ H5FD_s3comms_aws_canonical_request(char *canonical_request_dest, int _cr_size, c
     size_t      sh_size      = (size_t)_sh_size;
     size_t      cr_len       = 0; /* working length of canonical request str */
     size_t      sh_len       = 0; /* working length of signed headers str */
-    char        tmpstr[1024];
+    char        tmpstr[H5FD_ROS3_MAX_SECRET_TOK_LEN];
```

Comment:
I would allocate this dynamically.  4096 is probably big enough to start tripping our stack size warnings and we endeavor to be warning-free.